x-pack/plugins/index_management/server/routes/api/templates/register_get_routes.ts

```diff
@@ -18,40 +18,43 @@ import { RouteDependencies } from '../../../types';
 import { addBasePath } from '../index';
 
 export function registerGetAllRoute({ router, lib: { isEsError } }: RouteDependencies) {
-  router.get({ path: addBasePath('/index_templates'), validate: false }, async (ctx, req, res) => {
-    const { callAsCurrentUser } = ctx.dataManagement!.client;
-
-    try {
-      const cloudManagedTemplatePrefix = await getCloudManagedTemplatePrefix(callAsCurrentUser);
-
-      const legacyTemplatesEs = await callAsCurrentUser('indices.getTemplate');
-      const { index_templates: templatesEs } = await callAsCurrentUser(
-        'dataManagement.getComposableIndexTemplates'
-      );
-
-      const legacyTemplates = deserializeLegacyTemplateList(
-        legacyTemplatesEs,
-        cloudManagedTemplatePrefix
-      );
-      const templates = deserializeTemplateList(templatesEs, cloudManagedTemplatePrefix);
-
-      const body = {
-        templates,
-        legacyTemplates,
-      };
-
-      return res.ok({ body });
-    } catch (error) {
-      if (isEsError(error)) {
-        return res.customError({
-          statusCode: error.statusCode,
-          body: error,
-        });
+  router.get(
+    { path: addBasePath('/index_templates'), validate: false },
+    async (context, request, response) => {
+      const { client } = context.core.elasticsearch;
+
+      try {
+        const cloudManagedTemplatePrefix = await getCloudManagedTemplatePrefix(client);
+
+        const { body: legacyTemplatesEs } = await client.asCurrentUser.indices.getTemplate();
+        const {
+          body: { index_templates: templatesEs },
+        } = await client.asCurrentUser.indices.getIndexTemplate();
+
+        const legacyTemplates = deserializeLegacyTemplateList(
+          legacyTemplatesEs,
+          cloudManagedTemplatePrefix
+        );
+        const templates = deserializeTemplateList(templatesEs, cloudManagedTemplatePrefix);
+
+        const body = {
+          templates,
+          legacyTemplates,
+        };
+
+        return response.ok({ body });
+      } catch (error) {
+        if (isEsError(error)) {
```

Comment:
Can we use the `handleEsError` util here instead?

Reply:
I think this might be the only case where we're still using `isEsError`, so if we're able to remove it here, we should also be able to remove it from `shared_imports` and the api routes setup.

Reply:
Great idea! Done.